File: Day 3/Part 1.js
const fs = require('fs');
const sacks = fs.readFileSync('./Day 3/input.txt', 'utf8').split('\n')
let total = 0;

function doubles(a, b) {
    return b.filter(i => new Set(a).has(i));
}

sacks.forEach(i => {

    const conA = i.slice(0, i.length / 2);
    const conB = i.slice(i.length / 2);

    let letter = doubles(conA.split(''), conB.split(''));
    let letters = [...Array(26).keys()].map(i => String.fromCharCode(i + 97));
    let Letters = letters.concat(letters.map(i => i.toUpperCase()));

    total += (letter.map(sacks => Letters.indexOf(sacks) + 1)[0])
});


console.log(total);

File: Day 4/Part 1.py
with open('./Day 4/input.txt','r') as f:
  data = f.read().split("\n")

contain = 0

for line in data:
  a, b = line.split(",")
  x, y = map(int, a.split("-"))
  z, w = map(int, b.split("-"))
  
  if x >= z and y <= w or z >= x and w <= y:
    contain += 1

print(contain)